Sources/BraintreeCore/Configuration/ConfigurationLoader.swift

```diff
@@ -35,56 +36,33 @@ class ConfigurationLoader {
     ///   - `BTConfiguration?`: The configuration object if it is successfully fetched or retrieved from the cache.
     ///   - `Error?`: An error object if fetching the configuration fails or if the instance is deallocated.
     @_documentation(visibility: private)
-    func getConfig(completion: @escaping (BTConfiguration?, Error?) -> Void) {
+    func getConfig() async throws -> BTConfiguration {
+        if let existingTask {
+            return try await existingTask.value
+        }
+
         if let cachedConfig = try? configurationCache.getFromCache(authorization: http.authorization.bearer) {
-            completion(cachedConfig, nil)
-            return
+            return cachedConfig
         }
-        
-        pendingCompletions.add(completion)
-        
-        // If this is the 1st `v1/config` GET attempt, proceed with firing the network request.
-        // Otherwise, there is already a pending network request.
-        if pendingCompletions.count == 1 {
-            http.get(configPath, parameters: BTConfigurationRequest()) { [weak self] body, response, error in
-                guard let self else {
-                    self?.notifyCompletions(nil, BTAPIClientError.deallocated)
-                    return
-                }
-                
-                if let error {
-                    notifyCompletions(nil, error)
-                    return
-                } else if response?.statusCode != 200 || body == nil {
-                    notifyCompletions(nil, BTAPIClientError.configurationUnavailable)
-                    return
+     
+        let task = Task {
+            do {
```

Comment:
do we need this do catch block? It seems to be catching errors that are already thrown.